core/common/src/main/java/alluxio/extensions/ExtensionFactoryRegistry.java

```diff
@@ -270,4 +269,25 @@ private void unregister(T factory, List<T> factories) {
     LOG.debug("Unregistered factory implementation {} - {}", factory.getClass(), factory);
     factories.remove(factory);
   }
+
+  /**
+   * Selects all the factories that support the given path from the given list of factories.
+   *
+   * @param path path
+   * @param conf configuration of the extension
+   * @param factories list of factories
+   * @return list of factories that support the given path which may be an empty list
+   */
+  public List<T> select(String path, S conf, List<T> factories) {
+    Preconditions.checkArgument(path != null, "path may not be null");
```

Comment:
alluxio code convention:
```suggestion
    Preconditions.checkNotNull(path, "path");
```